### PR TITLE
Update rflink.markdown anchors and `ha_category`

### PR DESCRIPTION
**Pull request by javicalle**
Small markdown fix to add RFLink to categories and fix the links anchors

## Proposed change

Just noticied that the RFLink integration was not listed in the HA categories for generic devices (switch, light, cover,...).
I have updated the `ha_category` and fixed the page anchors.

## Type of change

- [x] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

## Summary by CodeRabbit

- **New Features**
	- Expanded device support in RFLink integration documentation to include Cover, Binary sensor, Light, Sensor, and Switch.
- **Documentation**
	- Improved clarity and consistency in configuration options and examples.
	- Enhanced maintainability with relative links.
	- Clarified behavior regarding automatic addition of devices, specifying default treatment of switchable devices as lights.
	- Added a section detailing device support tested with specific hardware (Nodo RFLink Gateway V1.4/RFLink R46).
	- Structural changes for better navigation and organization.

### CODE REVIEW

**Comment by javicalle** [comment]
I think all of these suggestions from coderabbitai are creating a mess.
I've checked other integrations sites and it doesn't seem like "header IDs" are commonly used. I'm going to remove all I can and see what happens.

File: source/_integrations/rflink.markdown [hunk 1]
```diff
@@ -137,7 +145,7 @@ sensor:
     automatic_add: true
 ```
 
-[RFLink Switches](/integrations/switch.rflink/) and [RFLink Binary Sensors](/integrations/binary_sensor.rflink/) cannot be added automatically.
+[RFLink Switches](#switch) and [RFLink Binary Sensors](#binary-sensor) cannot be added automatically.
```
[the remark this answers is not in the document]
Comment (javicalle) [reply source/_integrations/rflink.markdown:149]: Checked against the preview deploy and it's fine:
* https://deploy-preview-35925--home-assistant-docs.netlify.app/integrations/rflink/

File: source/_integrations/rflink.markdown [hunk 2]
```diff
@@ -137,7 +146,7 @@ sensor:
...
```
[the remark this answers is not in the document]
Comment (javicalle) [reply source/_integrations/rflink.markdown:149]: Checked against the preview deploy and it's fine:
- https://deploy-preview-35925--home-assistant-docs.netlify.app/integrations/rflink/

File: source/_integrations/rflink.markdown [hunk 3]
```diff
@@ -259,7 +264,7 @@ devices:
           required: false
           type: list
         device_class:
-          description: Sets the [class of the device](/integrations/binary_sensor/), changing the device state and icon that is displayed on the frontend.
+          description: Sets the [class of the device](/integrations/binary_sensor/#device-class), changing the device state and icon that is displayed on the frontend.
```
[the remark this answers is not in the document]
Comment (javicalle) [reply source/_integrations/rflink.markdown:267]: Done

File: source/_integrations/rflink.markdown (hunk 2 above)
[the remark this answers is not in the document]
Comment (javicalle) [reply source/_integrations/rflink.markdown:149]: Checked against the preview deploy and it's fine:

- https://deploy-preview-35925--home-assistant-docs.netlify.app/integrations/rflink/

File: source/_integrations/rflink.markdown [hunk 4]
```diff
@@ -259,7 +264,7 @@ devices:
           required: false
           type: list
         device_class:
-          description: Sets the [class of the device](/integrations/binary_sensor/), changing the device state and icon that is displayed on the frontend.
+          description: Sets the [class of the device](../binary_sensor/#device-class), changing the device state and icon that is displayed on the frontend.
```
Comment (frenck) [inline source/_integrations/rflink.markdown]: This change isn't correct. We don't use relative links in our documentation.

```suggestion
          description: Sets the [class of the device](/integrations/binary_sensor/#device-class), changing the device state and icon that is displayed on the frontend.
```

**Review by frenck — APPROVED** [review]
Thanks, @javicalle 👍 

../Frenck

**Pull request merged**